zerver/actions/streams.py

```diff
@@ -207,6 +218,14 @@ def do_unarchive_stream(
         is_web_public=None
     )
 
+    # Unset the is_realm_public cache on attachments, since the stream is now private.
+    Attachment.objects.filter(messages__recipient_id=stream.recipient_id).update(
+        is_realm_public=None
+    )
+    ArchivedAttachment.objects.filter(messages__recipient_id=stream.recipient_id).update(
+        is_realm_public=None
+    )
+
```

Comment:
There's no reason to not combine this with the previous `.update()`s right above this.